File: test/testscala/src/main/scala/com/test/scala/domain/TableOneDomain.scala
package com.test.scala.domain

import java.util
import java.util.Optional

import com.test.scala.entity.TableOne
import com.test.scala.repository.TableOneRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.jpa.domain.Specification
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

/**
  * @author zhangbin by 02/05/2018 17:43
  * @since JDK 11
  */
@Service
@Transactional
class TableOneDomain @Autowired()(private val tableOneRepository: TableOneRepository) {

  @Transactional
  def create(tableOne: TableOne): TableOne = tableOneRepository.save(tableOne)

  def query(name: String): Optional[TableOne] = tableOneRepository.findByName(name)

  def all(specification: Specification[TableOne]): util.List[TableOne] = tableOneRepository.findAll(specification)

}

File: test/testkotlin/src/main/kotlin/com/test/kotlin/domain/TableOneDomain.kt
package com.test.kotlin.domain

import com.test.kotlin.entity.TableOne
import com.test.kotlin.repository.TableOneRepository
import org.springframework.data.jpa.domain.Specification
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.util.*

/**
 * @author zhangbin by 28/04/2018 13:07
 * @since JDK 11
 */
@Service
@Transactional
class TableOneDomain(private val tableOneRepository: TableOneRepository) {

    @Transactional
    fun create(tableOne: TableOne): TableOne = tableOneRepository.save(tableOne)

    fun query(name: String): Optional<TableOne> = tableOneRepository.findByName(name)

    fun all(specification: Specification<TableOne>?): MutableList<TableOne> = tableOneRepository.findAll(specification)
}